src/AutoRest.CSharp.V3/Generation/Writers/CodeWriter.cs

```diff
@@ -376,13 +376,32 @@ public void RemoveTrailingComma()
                     continue;
                 }
 
+                action(i);
+
+                break;
+            }
+        }
+
+        public void RemoveTrailingComma()
+        {
+            ActionOnLast(i =>
+            {
                 if (_builder[i] == ',')
                 {
                     _builder.Remove(i, _builder.Length - i);
                 }
+            });
+        }
 
-                break;
-            }
+        private void CreateNewLine()
```

Comment:
Why don't you have this logic in Line method itself.

Reply:
Done.